prow/git/v2/executor.go

```diff
@@ -0,0 +1,72 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package git
+
+import (
+	"os/exec"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+)
+
+// Executor knows how to execute Git commands
+type Executor interface {
+	Run(args ...string) ([]byte, error)
+}
+
+func NewCensoringExecutor(dir string, censor func(content []byte) []byte, logger *logrus.Entry) (Executor, error) {
```

Comment:
Are we planning on ever adding a non-censoring executor? If not, maybe rename to just `NewExecutor`?